clang/include/clang/Sema/Sema.h

```diff
@@ -12836,6 +12834,32 @@ class Sema final {
   }
 };
 
+template <typename AttrType>
+void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
+                                             const AttributeCommonInfo &CI,
+                                             Expr *E) {
+  if (!E)
+    return;
```

Comment:
Should this be an assert?  When is this allowable?

I see that it should have already asserted if !E, so an assert seems more reasonable here.

Reply:
Yes, i have added assert here.

Reply:
@Erich, several codegen tests are failing like this after adding assertion here:

 /export/iusers/smanna/my_sycl_ws/llvm/clang/include/clang/Sema/Sema.h:12841: void clang::Sema::addIntelSYCLSingleArgFunctionAttr(clang::Decl*, const clang::AttributeCommonInfo&, clang::Expr*) [with AttrType = clang::IntelReqdSubGroupSizeAttr]: Assertion `!E' failed.

Reply:
Assert should be that E exists. Right now you're asserting !E

Reply:
Actually no. I was confused. Disregard my previous comment

Reply:
Disregard @elizabethandrews LAST message, her first one was right :)

Reply:
Lol yes. I need a break :D

Reply:
> Assert should be that E exists. Right now you're asserting !E
Yes, i was in wrong way. Sorry about this. Thanks Elizabeth.